store/tikv/coprocessor.go

```diff
@@ -456,84 +465,164 @@ func (it *copIterator) Next() ([]byte, error) {
 	return resp.Data, nil
 }
 
-// handleTask handles single copTask.
-func (it *copIterator) handleTask(bo *Backoffer, task *copTask) []copResponse {
-	coprocessorCounter.WithLabelValues("handle_task").Inc()
+// handleTasks handles single copTask, sends the result to channel, retry automatically on error.
+func (it *copIterator) handleTask(goCtx goctx.Context, bo *Backoffer, task *copTask, ch chan copResponse) {
+	var buf = [1]*copTask{task}
+	remainTasks := buf[:1]
```

Comment:
why not `remainTasks := []*copTask{task}` directly?

Reply:
An optimization to make less object allocation, at the cost of readability, I'll change here.